obspy/taup/tau_model.py

```diff
@@ -413,3 +371,206 @@ def getBranchDepths(self):
+        # c) handle simple contents of .sMod
+        dtypes = [(native_str('DEBUG'), np.bool_),
+                  (native_str('DEFAULT_SLOWNESS_TOLERANCE'), np.float_),
+                  (native_str('PWAVE'), np.bool_),
+                  (native_str('SWAVE'), np.bool_),
+                  (native_str('allowInnerCoreS'), np.bool_),
+                  (native_str('maxDeltaP'), np.float_),
+                  (native_str('maxDepthInterval'), np.float_),
+                  (native_str('maxInterpError'), np.float_),
+                  (native_str('maxRangeInterval'), np.float_),
+                  (native_str('minDeltaP'), np.float_),
+                  (native_str('radiusOfEarth'), np.float_),
+                  (native_str('slowness_tolerance'), np.float_)]
+        slowness_model = np.empty(shape=(), dtype=dtypes)
+        for dtype in dtypes:
+            key = dtype[0]
+            slowness_model[key] = getattr(self.sMod, key)
+        arrays['sMod'] = slowness_model
+        # d) handle complex contents of .sMod
+        arrays['sMod.PLayers'] = self.sMod.PLayers
+        arrays['sMod.SLayers'] = self.sMod.SLayers
+        arrays['sMod.criticalDepths'] = self.sMod.criticalDepths
+        for key in ['fluidLayerDepths', 'highSlownessLayerDepthsP',
+                    'highSlownessLayerDepthsS']:
+            data = getattr(self.sMod, key)
+            if len(data) == 0:
+                arr_ = np.array([])
+            else:
+                arr_ = np.vstack([data_._to_array() for data_ in data])
+            arrays['sMod.' + key] = arr_
+        # e) handle .sMod.vMod
+        dtypes = [(native_str('cmbDepth'), np.float_),
+                  (native_str('default_cmb'), np.float_),
+                  (native_str('default_iocb'), np.float_),
+                  (native_str('default_moho'), np.int_),
+                  (native_str('iocbDepth'), np.float_),
+                  (native_str('isSpherical'), np.bool_),
+                  (native_str('maxRadius'), np.float_),
+                  (native_str('minRadius'), np.int_),
+                  (native_str('modelName'), np.str_,
+                   len(self.sMod.vMod.modelName)),
+                  (native_str('mohoDepth'), np.float_),
+                  (native_str('radiusOfEarth'), np.float_)]
+        velocity_model = np.empty(shape=(), dtype=dtypes)
+        for dtype in dtypes:
+            key = dtype[0]
+            velocity_model[key] = getattr(self.sMod.vMod, key)
+        arrays['vMod'] = velocity_model
+        arrays['vMod.layers'] = self.sMod.vMod.layers
+        # finally save the collection of (structured) arrays to binary file
+        np.savez_compressed(filename, **arrays)
+
+    @staticmethod
+    def deserialize(filename):
+        """
+        Deserialize model from numpy npz binary file.
+        """
+        with np.load(filename) as npz:
+            model = TauModel(sMod=None, skip_calc=True)
+            # a) handle simple contents
+            complex_contents = [
```

Comment:
simple or complex? These two lines _seem_ contradictory (though not actually.) Maybe just move the comment below this variable.